### PR TITLE
Fix sharing

### LINKED ISSUE #608: FileMetadata broken on android
Android require an other version adapter

### PR DESCRIPTION
**Pull request by JoGir**
Resolves issues for android integration
fixes #608

### CODE REVIEW

File: box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt [hunk 1]
```diff
@@ -5,12 +5,21 @@ import de.qabel.box.storage.BoxFile
...
 import java.io.File
 import java.io.IOException
+import java.sql.Connection
 import java.sql.DriverManager
 import java.sql.SQLException
 
 class JdbcFileMetadataFactory(val tmpDir: File) : FileMetadataFactory {
+
+    companion object {
+        var versionAdapterFactory : (connection : Connection) -> VersionAdapter = { PragmaVersionAdapter(it)}
```
Comment (julianseeger) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: Can't this be a constructor argument?

Comment (audax) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: Could be done in the same way as with the DirectoryMetadata

Comment (JoGir) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: and the sharingService requires a VersionAdapter then? i confirm this isnt a nice way, but this needs some more refactoring then change this to the constructor...

Comment (audax) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: The SharingService then requires a FileMetadataFactory which on android must be configured to use the android version adapter

Comment (audax) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: the factory can default to the factory from the core and with @JVMOverloads the default constructor wouldn't change. On Android you can pass the factory in from dagger.

Comment (julianseeger) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: What's wrong with constructor injection?

Comment (audax) [inline box/src/main/java/de/qabel/box/storage/jdbc/JdbcFileMetadataFactory.kt]: It now works with constructor injection

**Pull request merged**